instrumentation/rocketmq-client-4.8/testing/src/main/java/base/IntegrationTestBase.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package base;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.rocketmq.broker.BrokerController;
+import org.apache.rocketmq.common.BrokerConfig;
+import org.apache.rocketmq.common.UtilAll;
+import org.apache.rocketmq.common.namesrv.NamesrvConfig;
+import org.apache.rocketmq.logging.InternalLogger;
+import org.apache.rocketmq.logging.InternalLoggerFactory;
+import org.apache.rocketmq.namesrv.NamesrvController;
+import org.apache.rocketmq.remoting.netty.NettyClientConfig;
+import org.apache.rocketmq.remoting.netty.NettyServerConfig;
+import org.apache.rocketmq.store.config.MessageStoreConfig;
+import org.apache.rocketmq.test.util.MQAdmin;
+import org.apache.rocketmq.test.util.TestUtils;
+import org.junit.Assert;
+
+public class IntegrationTestBase {
+  public static final InternalLogger logger =
+      InternalLoggerFactory.getLogger(IntegrationTestBase.class);
+
+  protected static final String SEP = File.separator;
+  protected static final String BROKER_NAME_PREFIX = "TestBrokerName_";
+  protected static final AtomicInteger BROKER_INDEX = new AtomicInteger(0);
+  protected static final List<File> TMPE_FILES = new ArrayList<>();
+  protected static final List<BrokerController> BROKER_CONTROLLERS = new ArrayList<>();
+  protected static final List<NamesrvController> NAMESRV_CONTROLLERS = new ArrayList<>();
+  protected static int topicCreateTime = 30 * 1000;
+  protected static final int COMMIT_LOG_SIZE = 1024 * 1024 * 100;
+  protected static final int INDEX_NUM = 1000;
+  private static final AtomicInteger port = new AtomicInteger(40000);
+
+  public static synchronized int nextPort() {
+    return port.addAndGet(random.nextInt(10) + 10);
+  }
+
+  protected static final Random random = new Random();
+
+  public static String createBaseDir() {
+    String baseDir = System.getProperty("user.home") + SEP + "unitteststore-" + UUID.randomUUID();
```

Comment:
How about using `Files.createTempDirectory()` instead? Or using the JUnit `TemporaryFolder`/`TempDir` rule